feature-libs/customer-ticketing/core/http-interceptors/handlers/bad-ticket-request.handler.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpErrorResponse, HttpRequest } from '@angular/common/http';
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import {
+  ErrorModel,
+  GlobalMessageType,
+  HttpErrorHandler,
+  HttpResponseStatus,
+  Priority,
+  RoutingService,
+  GlobalMessageService,
+} from '@spartacus/core';
+import { isTicketNotFoundError } from '../../utils/utils';
+@Injectable({
+  providedIn: 'root',
+})
+export class BadTicketRequestHandler extends HttpErrorHandler {
```

Comment:
Thank you, @bgambocjaviniar .. The intention was to have a placeholder for handling other bad request error as well, including the `ticketNotFound` error.

Reply:
Thanks for the comment! But I do recall sharing an example to derive it from, which is `feature-libs/cart/base/core/http-interceptors/handlers/bad-cart-request.handler.ts` 

In this sense, 'bad request' for cart makes sense since it's a handler specifically for `400 - bad requests`. Similarly, this class for the ticket handler is targeting specifically the `404 - not found error`. Currently, they  are 1 to 1 in terms of handler creation and what it handles

Reply:
Thank you so much @bgambocjaviniar for your comment.. But the backend throws other errors too including 400 for customer ticketing. It is true that we do not have any bad request other than ticket not found, but if we choose to handle other bad request errors, they can be included here in this same class, keeping the similarity with other modules, like the 'cart'. I can surely change the class name, if you still think so.

Reply:
We will discuss this during our call for your recent push for fixing the error that determines whether or not it's a not found or not.



-----------
For simplicity, an example of what I'm mentioning is that these handlers have a single responsibility. Either it handles a not found request, a bad request, a forbidden request, etc. 

As seen in the screenshot, we inject several handlers, and customers can as well. As explained and seen in the code, the specific 'handling' is specifically looking only at the '404' errors. This is a 1-1 relationship and how it's being made and handled. 

In the end, anyone can inject several handlers for 'x' different errors. But the case you're currently looking at is specifically `404`

![Screenshot 2023-01-26 at 11 26 10 AM](https://user-images.githubusercontent.com/29099466/214891874-1f4b5bee-f65a-49d0-994c-588db63f9e97.png)